src/controller/transaction-controller.ts

```diff
@@ -157,8 +157,7 @@ export default class TransactionController extends BaseController {
 
       // verify balance if user cannot have negative balance.
       const user = await User.findOne({ where: { id: body.from } });
-      const allowNegative = this.roleManager.can(await this.roleManager.getRoles(user), 'update', 'own', 'Balance', ['negative']);
-      if (!allowNegative && !await TransactionService.verifyBalance(body)) {
+      if (!user.canGoIntoDebt && !await TransactionService.verifyBalance(body)) {
```

Comment:
Is this preferred? Now going into debt is no longer role based, which seems less flexible.

Reply:
Currently, there are no mechanisms to dynamically assign roles to and remove roles from users. If we wanted to use RBAC, we have to create these. Also, when using RBAC, in theory multiple roles can define whether someone can go into debt. Finding out based on the list of roles whether one can go into debt is therefore much harder. Simply creating a boolean that is related to the user entity, is therefore in my opinion the most simple solution. We can then also reuse the existing `createUser` and `updateUser` endpoints.